### ISSUE TITLE
Fanboy's Social Blocking List

### ISSUE THREAD

**Issue by belodelokelo** [nextdns/metadata#943]
Hello NextDNS Team. Could you add the "Fanboy's Social Blocking List?" I have noticed that the "Social media filter" already on NextDNS doesn't do the job. 

https://easylist.to

**Comment by quantumpacket** [comment]
@belodelokelo if you're referring to this https://easylist.to/easylist/fanboy-social.txt it will not work with NextDNS. That is meant for a browser addon to consume and is not meant for DNS.

**Comment by belodelokelo** [comment]
Is there any other available that can work with DNS?

**Comment by SohanRay** [comment]
I think adguard DNS host and easylist are the only ones. The level of blocking that you are looking for might not be possible for a DNS, but possible only for browser ad ons. Just like DNS cannot block Youtube ads but browser ad ons can.

**Comment by quantumpacket** [comment]
No, the currently available list https://easylist.to/easylist/easylist.txt is used partially by NextDNS, but it only parses and uses the absolute domain rules as far as I am aware. Otherwise it leads to false positives as DNS cannot have fine grained control over a request such as a browser addon.

So for example:

The rule `||shusacem.net^` can be used to block `shusacem.net` as the rule is for a specific hostname. However, `webtoolhub.com###banner_704_134` shouldn't be used to block `webtoolhub.com` as the rule is to block a specific element on the website, and is not meant for block the domain entirely. If you require blocking elements on a page you need to use a browser addon such as uBlock Origin. The fanboy social list you are wanting consists of element blocking rules, so it cannot be used for DNS blocking.

**Comment by belodelokelo** [comment]
Totally got it. Thank you. 

Basically all these are cosmetic filtering rather than actual calling of urls. Right?

If this is the case, should I use 1Blocker and activate only Hidden Element blocking instead of Block URLs since the latter is done from DNS?

**Comment by SohanRay** [comment]
yes,DNS can only block specific domains. Rest , elements can be blocked by browser extensions.

**Comment by quantumpacket** [comment]
@belodelokelo No, I would still keep it to block domains. As the browser addon will block the domain instantly. Not doing so, will cause the browser to make a DNS request for each otherwise blocked domain, which NexDNS will respond back with 0.0.0.0. This unnecessary DNS request may or may not have a noticeable overhead depending on how many domains the page is querying and how fast your connection to the NextDNS server is.

**Comment by belodelokelo** [comment]
> @belodelokelo No, I would still keep it to block domains. As the browser addon will block the domain instantly. Not doing so, will cause the browser to first make a DNS request for each otherwise blocked domain, which NexDNS will respond back with 0.0.0.0. This useless request may or may not have a noticeable overhead depending on how many domains the page is querying, how fast your connection to the NextDNS server is, among other factors. If the downloaded blocklist in the browser already knows to block a domain, why make a needless DNS request to determine the same thing?

Thank you @quantumpacket. But if that's the case then what's the point of blocking through DNS? Isn't it supposed to make things faster if I block through DNS?

**Comment by quantumpacket** [comment]
> Thank you @quantumpacket. But if that's the case then what's the point of blocking through DNS? Isn't it supposed to make things faster if I block through DNS?

Of course, it makes it faster by not allowing the connection to actually download the content being requested by never making a direct connection to the server's IP address. It basically black holes the request by pointing the connection to 0.0.0.0. You can speed things up a bit more by increasing your cache TTL, which will reduce the amount of repeated DNS requests for the same domain.

**Comment by belodelokelo** [comment]
1/ To increase cache TTL I put ON the setting "Cache Boost?"

2/ Regarding DNS blocking and content blocking with 1Blocker what is the best setting? To have blocklists activated on NextDNS and also block through 1Blocker the same things?

**Comment by quantumpacket** [comment]
1) Yes, I'm not sure what the default TTL is, but if you have NextDNS on your router you can also set it via the `max_ttl` setting.

2) I would keep the blocklists on both. This will allow NextDNS to cover all devices and not just browsers, the browser will still be able to block elements, but at the same time not make needless DNS requests. Because let's be real the most DNS requests being made are likely going to be through your browser. So let 1Blocker block them immediately, instead of needing to spend the time querying NextDNS.

**Comment by belodelokelo** [comment]
1/ You mean to enable this right? <img width="468" alt="Screen Shot 2022-03-07 at 14 33 39" src="https://user-images.githubusercontent.com/98962550/157044122-069f6a89-46b6-4e43-8ede-cfec1b69bbcb.png">

2/ Thank you. Regarding only the browser and not other apps/devices, if I have 1Blocker enabled does with same blocklists enabled, make the sites load faster? I am asking because in general with dns blocking the browser is also less streessed due to not handling many lists from content blocking.

Please let me know and thank. Waiting for your repose and I will close the issue.

**Comment by quantumpacket** [comment]
1. Yes, what that does is set a TTL for the DNS response. So if the TTL is say 60 seconds, any requests to the same domain will use the stored DNS response without making a new DNS request to NextDNS for 60 seconds, after TTL expiration it will make a new DNS request and repeat the process. This helps reduce DNS requests for domains that are needing to be queried multiple times within the TTL timeframe.

2. The browser can easily handle the blocklists as the lists are already downloaded and loaded. A DNS request on the other hand would take up more resources as it needs to make a connection to the DNS server, then wait for a response back. Imagine you had 10 bad domains being loaded on a webpage. 1Blocker with its downloaded lists will block them immediately. Otherwise your browser will make 10 different DNS requests, and have to wait for 10 different responses.

The below is a single DNS request. My query time was 99ms, which can vary based on network conditions. So those 10 DNS requests could have become 990ms of extra time added, when 1Blocker would have blocked them immediately. Less requests/connections are better.

> $ dig xbox.com     
>
> ; <<>> DiG  <<>> xbox.com
> ;; global options: +cmd
> ;; Got answer:
> ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 1163
> ;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
> 
> ;; QUESTION SECTION:
> ;xbox.com.                      IN      A
> 
> ;; ANSWER SECTION:
> xbox.com.               5       IN      A       0.0.0.0
> 
> ;; Query time: 99 msec
> ;; SERVER: x.x.x.x#53(x.x.x.x)
> ;; WHEN: Mon Mar 07 08:56:53 EST 2022
> ;; MSG SIZE  rcvd: 42

**Comment by krystian3w** [comment]
https://github.com/nextdns/metadata/issues/943#issuecomment-1059994700 - So NextDNS dropped convert:

    ||example.org^$popup
    ||example.com^$thirdparty

To dns blockades?

**Comment by hagezi** [comment]
@krystian3w
If you also extract the rules with `$popup` and `$third-party` for domain lists from AdBlock lists, which were actually created for content blockers, you are guaranteed to get false positive domains. These should not be blindly applied to DNS lists.

How the NextDNS parser handles this, I don't know.

I don't use them for my DNS blocklists in any case.

Greetings,
Gerd - https://github.com/hagezi/dns-blocklists

**Comment by krystian3w** [comment]
I think still import domain between `||` & `^` and waiting to report real False Positive.
